crates/bevy_input/src/axis.rs

```diff
@@ -21,15 +25,113 @@ impl<T> Axis<T>
 where
     T: Copy + Eq + Hash,
 {
-    pub fn set(&mut self, axis: T, value: f32) -> Option<f32> {
-        self.axis_data.insert(axis, value)
+    /// Inserts a position data for an input device,
+    /// restricting the position data to an interval `min..=max`.
+    ///
+    /// If the input device wasn't present before, [None] is returned.
+    ///
+    /// If the input device was present, the position data is updated, and the old value is returned.
+    pub fn set(&mut self, input_device: T, position_data: f32) -> Option<f32> {
+        let new_position_data = position_data.clamp(self.get_min(), self.get_max());
+        self.axis_data.insert(input_device, new_position_data)
     }
 
-    pub fn get(&self, axis: T) -> Option<f32> {
-        self.axis_data.get(&axis).copied()
+    /// Returns a position data corresponding to the input device.
+    pub fn get(&self, input_device: T) -> Option<f32> {
+        self.axis_data.get(&input_device).copied()
     }
 
-    pub fn remove(&mut self, axis: T) -> Option<f32> {
-        self.axis_data.remove(&axis)
+    /// Removes the position data of the input device,
+    /// returning the position data if the input device was previously set.
+    pub fn remove(&mut self, input_device: T) -> Option<f32> {
+        self.axis_data.remove(&input_device)
+    }
+
+    /// Returns maximum allowed position data, which is 1.0.
+    pub fn get_min(&self) -> f32 {
```

Comment:
This is an "accessor-style" method and should use the convention `min()` instead of `get_min()`